modules/tinymce/src/themes/silver/main/ts/ui/sizing/EditorSize.ts

```diff
@@ -5,17 +5,43 @@ import Editor from 'tinymce/core/api/Editor';
 import * as Options from '../../api/Options';
 import * as Utils from './Utils';
 
+const convertValueToPx = (editor: Editor, value: number | string): Optional<number> => {
+  if (typeof value === 'number') {
+    return Optional.from(value);
+  }
+
+  return Utils.parseToInt(value.trim()).orThunk(() => {
+    const splitValue = /^([0-9.]+)(pt|em|px)$/.exec(value);
+
+    if (splitValue) {
+      const type = splitValue[2];
+      const parsed = Number.parseFloat(splitValue[1]);
+
+      if (Number.isNaN(parsed) || parsed < 0) {
+        return Optional.none();
+      } else if (type === 'em') {
+        return Optional.from(parsed * Number.parseFloat(window.getComputedStyle(editor.targetElm).fontSize));
```

Comment:
This is the only time editor is used. So you could lift this out to a module that take just an element so that it's generic maybe put it in the Utils module next to parseToInt. Maybe a SugarElement then you pass that in instead. The benefit of that is that it becomes easier to test in isolation and we could in theory use this for other convert calls in the future seems like a useful thing to have.

Reply:
I agree it'd be good to move it elsewhere, I just wasn't sure where to ( FontUtils felt the most likely but it didn't fit, so I didn't want to put it there ). This seems like a decent solution.